components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -431,23 +432,53 @@ - (CGFloat)justifiedWidth {
 - (void)scrollUntilSelectedItemIsVisibleWithoutAnimation {
   NSUInteger index = [self.items indexOfObject:self.selectedItem];
   if (index == NSNotFound || index >= self.containerView.arrangedSubviews.count) {
+    index = 0;
+  }
+  if (self.containerView.arrangedSubviews.count == 0U) {
     return;
   }
 
-  CGFloat selectedItemOriginX = 0;
+  CGRect estimatedItemFrame = [self estimatedFrameInContainerViewForItemAtIndex:index];
+  [self scrollRectToVisible:[self convertRect:estimatedItemFrame fromView:self.containerView]
+                   animated:NO];
+}
+
+- (CGRect)estimatedFrameInContainerViewForItemAtIndex:(NSUInteger)index {
+  if (index == NSNotFound || index >= self.containerView.arrangedSubviews.count) {
+    return CGRectZero;
+  }
+
+  BOOL isRTL =
+      [self mdf_effectiveUserInterfaceLayoutDirection] == UIUserInterfaceLayoutDirectionRightToLeft;
+  CGFloat viewOriginX =
+      isRTL ? CGRectGetMaxX(self.containerView.bounds) : CGRectGetMinX(self.containerView.bounds);
+
   for (NSUInteger i = 0; i < index; ++i) {
-    CGSize expectedItemSize = [self expectedSizeForView:self.containerView.arrangedSubviews[i]];
-    selectedItemOriginX += expectedItemSize.width;
+    CGSize viewSize = [self expectedSizeForView:self.containerView.arrangedSubviews[i]];
+    if (isRTL) {
+      viewOriginX -= viewSize.width;
+    } else {
+      viewOriginX += viewSize.width;
+    }
   }
-  CGSize expectedSelectedItemSize =
-      [self expectedSizeForView:self.containerView.arrangedSubviews[index]];
-  CGRect expectedItemFrame =
-      CGRectMake(CGRectGetMinX(self.bounds) + selectedItemOriginX, CGRectGetMinY(self.bounds),
-                 expectedSelectedItemSize.width, expectedSelectedItemSize.height);
-  [self scrollRectToVisible:expectedItemFrame animated:NO];
+  CGSize viewSize = [self expectedSizeForView:self.containerView.arrangedSubviews[index]];
+  if (isRTL) {
+    viewOriginX -= viewSize.width;
+  }
+  CGRect itemFrameInContainerView = CGRectMake(
+      viewOriginX, CGRectGetMinY(self.containerView.bounds), viewSize.width, viewSize.height);
+  CGRect convertedRect = [self convertRect:itemFrameInContainerView fromView:self.containerView];
+  return convertedRect;
 }
 
 - (CGSize)expectedSizeForView:(UIView *)view {
+  if (self.containerView.distribution == UIStackViewDistributionFillEqually) {
+    if (CGRectGetWidth(self.containerView.bounds) < 0) {
+      return CGSizeMake(
+          CGRectGetWidth(self.containerView.bounds) / self.containerView.arrangedSubviews.count,
+          CGRectGetHeight(self.containerView.bounds));
+    }
+  }
```

Comment:
Nit: Should the 2nd part of this method be inside an "else" to make it more clear that there are 2 cases for this method?

Reply:
Do you mean you'd like me to combine the two `if` conditions into a single expression?  I think early returns are preferred over indentation. @jverkoey has commented on multiple occasions (and I agree) that unnecessary indentation tends to make the code less readable.  At least it generally does.

Reply:
Combined the two conditions into a single. That removed one layer of indentation. I still feel that the early return is easier to read than keeping an `else` block. The `else` requires the reader to mentally check its start when reaching the end.